### PR TITLE
fix: test-ng: kernel module unload - `sg` issue

### LINKED ISSUE #3880: test-ng: test_sysdiff_after_tests finds `sg` kernel module
**What happened:**

The test_sysdiff_after_tests finds `sg` kernel module.

https://github.com/gardenlinux/gardenlinux/actions/runs/19332985908/job/55301813280?pr=3877

```
 _______________________________________________ test_sysdiff_after_tests _______________________________________________
test_sysdiff.py:49: in test_sysdiff_after_tests
    pytest.fail(
E   Failed: System changes were detected during the test run:
E   === Kernel module changes (2025-11-13T13:40:25.741508-root-before-tests -> 2025-11-13T13:42:01.899847-root-after-tests) ===
E   --- kernel_modules@snapshot_a
E   +++ kernel_modules@snapshot_b
E   @@ -48,6 +48,7 @@
E    scsi_common
E    scsi_mod
E    scsi_transport_iscsi
E   +sg
E    sha1_ce
E    sha256_arm64
E    sha2_ce
E   
E   Tip: lines starting with '+' mean new/changed; '-' mean removed/previous state.
```

This is most likely related to https://github.com/gardenlinux/gardenlinux/pull/3494 and
https://github.com/gardenlinux/gardenlinux/blob/85ca8d0e47df0c2c3cc68da6cf96313e64b5109d/tests-ng/handlers/iscsi.py#L67C19-L67C38

It seems the kernel module unload is not working as expected. It could also be a corner case or timing issue.

### PR DESCRIPTION
**Pull request by yeoldegrove**
**What this PR does / why we need it**:

This PR adds additional checks to the test-ng kernel module plugin to make sure modules are really unloaded before returning True. Make sure to only unload kernel modules that were loaded by the plugin and add debugging output to debug edge cases.
It removes the unsafe kernel module functions and adapts checks to use the new functionality in a consistent way.

**Which issue(s) this PR fixes**:
Fixes #3880

### CODE REVIEW

**Review by ByteOtter — APPROVED** [review]
to me this looks good 👍🏼
